client/internal/packagessyncer.go

```diff
@@ -290,7 +296,20 @@ func (s *packagesSyncer) downloadFile(ctx context.Context, pkgName string, file
 		return fmt.Errorf("cannot download file from %s, HTTP response=%v", file.DownloadUrl, resp.StatusCode)
 	}
 
-	err = s.localState.UpdateContent(ctx, pkgName, resp.Body, file.ContentHash, file.Signature)
+	// Package length is required to be able to report download percent.
+	packageLength := -1
+	if contentLength := resp.Header.Get("Content-Length"); contentLength != "" {
+		if length, err := strconv.Atoi(contentLength); err == nil {
+			packageLength = length
+		}
+	}
+	// start the download reporter
+	detailsReporter := newDownloadReporter(downloadReporterDefaultInterval, packageLength) // TODO set interval
```

Comment:
I need some guidance on where the user of a client library can be expected to set a value for the update interval, or if we even need to expose it

Reply:
I think this should be made customisable.

Reply:
Yes, my comment is more about my confusion as to where we should try to pass the config option from; Should we add `PackageReporterInterval/SetPackageReporterInterval) funcs to the `types.PackagesStateProvider` interface or does it belong somewhere else?

Reply:
I think it should be part of `StartSettings`

Reply:
`StartSettings` is not currently being passed to the `NewPackagesSyncer` function; do we want to add it as another parameter?

Reply:
NewPackagesSyncer is internal, so passing the interval param should be fine. This is similar to how we enabled heartbeat interval.